frontend/src/hooks/usePersistentTableState.ts

```diff
@@ -39,7 +39,41 @@ export const usePersistentTableState = <T extends QueryParamConfigMap>(
         queryParamsDefinition,
     );
 
-    const [tableState, setTableState] = useQueryParams(queryParamsDefinition);
+    const [tableState, setTableStateInternal] = useQueryParams(
+        queryParamsDefinition,
+    );
+
+    type SetTableStateInternalParam = Parameters<
+        typeof setTableStateInternal
+    >[0];
+
+    const setTableState = useCallback(
+        (newState: SetTableStateInternalParam) => {
+            if (!queryParamsDefinition.offset) {
+                return setTableStateInternal(newState);
+            }
+            if (typeof newState === 'function') {
```

Comment:
this is sth we don't use but the original library allows for passing fn instead of new state